src/domain/checks/DownloadRecordChecks.ts

```diff
@@ -0,0 +1,26 @@
+import { pipe } from 'fp-ts/function';
+import * as R from 'fp-ts/Reader';
+import * as RA from 'fp-ts/ReadonlyArray';
+import * as O from 'fp-ts/Option';
+import { isGetNotificationDocumentMetadataRecord } from '../GetNotificationDocumentMetadataRecord';
+import { isGetPaymentNotificationMetadataRecord } from '../GetPaymentNotificationMetadataRecord';
+import { isDownloadRecord } from '../DownloadRecord';
+
+export const calledDownloadEndpointC = pipe(
+  R.Do,
+  R.apS('getNotificationDocumentMetadataRecordList', RA.filterMap(isGetNotificationDocumentMetadataRecord)),
+  R.apS('getPaymentNotificationMetadataRecordList', RA.filterMap(isGetPaymentNotificationMetadataRecord)),
+  R.apS('downloadRecordList', RA.filterMap(isDownloadRecord)),
+  R.map(({ getNotificationDocumentMetadataRecordList, getPaymentNotificationMetadataRecordList, downloadRecordList }) =>
+    pipe(
+      RA.concatW(getNotificationDocumentMetadataRecordList)(getPaymentNotificationMetadataRecordList),
+      RA.filterMap(({ output }) => (output.statusCode === 200 ? O.fromNullable(output.returned.url) : O.none)),
+      RA.every((url) =>
```

Comment:
I also changed this method from `RA.exists` to `RA.every`. WDYT it is better @datalek?